pkgs/development/compilers/ghc/8.10.2-binary.nix

```
@@ -139,7 +145,7 @@ in
 stdenv.mkDerivation rec {
   inherit version;
 
-  name = "ghc-${version}-binary";
+  name = "ghc-${version}-binary${binDistUsed.variantSuffix}";
```

Appending something to a derivation `name` is usually a bad idea since `builtins.parseDrvName` interprets everything after the first number after a dash as the version.

We are already doing this for the binary ones, but we shouldn't expand this. So let's take this opportunity to redo the naming of the binary derivations as well and move the `-binary` suffix before the version number (maybe in a separate commit?):

```suggestion
  name = "ghc-binary${binDistUsed.variantSuffix}-${version}";
```

Of course would need to be reflected in the other derivations as well. If you don't feel like doing it, I can also take care of that for you.

---

That's a good point.

I think we should do it the way you suggest.

---

This was done in #139164 which I've already merged into `haskell-updates` (with the variant suffix changes as well).

---

This was done in PR #139164.